### PR TITLE
Include goto-cl regression tests in test sequence [depends-on: #3674]

### PR DESCRIPTION
**Pull request by tautschnig**
Instead of special-casing the goto-cl folder via the CI configuration move all
target-platform specific logic into the (C)Makefile.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by chrisr-diffblue — APPROVED** [review]
Assuming CI passes, this looks much cleaner - many thanks!

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: c9b1ae832f3db66b7ecee78f6b3274330d174bfc).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96163307

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: bca22076f6a96230130be329e5f192c80ea8fdcd).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96163777

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5050350ab77a3b66a953f5212befcf37cf814436).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/96168847

File: regression/cbmc/unsigned___int128/main.c [hunk 1]
```diff
@@ -1,3 +1,8 @@
+#if defined(_WIN32) && !defined(__CYGWIN__)
+void reduce()
```
Comment (kroening) [inline regression/cbmc/unsigned___int128/main.c]: I would check for the (absence of the) gcc define -- I believe __int128 is a gcc extension; so this isn't about Windows.

**Review by kroening — APPROVED** [review]

**Comment by tautschnig** [comment]
It seems we can get rid of the hack to remove directories in Windows builds once all the depended-on PRs are merged (with one yet to be created).

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: acc8d62dfb4736c7d0dd9b37d9dd66d301a981d3).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97016238

**Pull request merged**